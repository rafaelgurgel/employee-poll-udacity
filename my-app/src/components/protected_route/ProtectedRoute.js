import React from 'react';
import { Navigate } from 'react-router-dom';

export default function ProtectedRoute({ authedUser, children }) {
  if (!authedUser) {
    return <Navigate to="/" replace />;
  }
  return children;
}
