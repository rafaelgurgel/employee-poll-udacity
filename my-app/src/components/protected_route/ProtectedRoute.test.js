import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter, Routes, Route } from 'react-router-dom';
import ProtectedRoute from './ProtectedRoute';
import '@testing-library/jest-dom';

function MockComponent() {
  return <div>Mock Child Content</div>;
}

beforeAll(() => {
  jest.spyOn(console, 'warn').mockImplementation((message) => {
    if (!message.includes('React.startTransition')) {
      console.warn(message);
    }
  });
});

afterAll(() => {
  jest.restoreAllMocks();
});

describe('ProtectedRoute Component', () => {
  it('renders the child component when user is authenticated', () => {
    const { getByText } = render(
      <MemoryRouter initialEntries={['/']}>
        <Routes>
          <Route
            path="/"
            element={
              <ProtectedRoute authedUser="sarahedo">
                <MockComponent />
              </ProtectedRoute>
            }
          />
        </Routes>
      </MemoryRouter>
    );

    expect(getByText(/Mock Child Content/i)).toBeInTheDocument();
  });

  it('redirects to the login page when user is not authenticated', () => {
    const { queryByText, container } = render(
      <MemoryRouter initialEntries={['/protected']}>
        <Routes>
          <Route
            path="/protected"
            element={
              <ProtectedRoute authedUser={null}>
                <MockComponent />
              </ProtectedRoute>
            }
          />
          <Route path="/" element={<div>Login Page</div>} />
        </Routes>
      </MemoryRouter>
    );

    expect(queryByText(/Mock Child Content/i)).not.toBeInTheDocument();

    expect(container).toHaveTextContent(/Login Page/i);
  });
});
