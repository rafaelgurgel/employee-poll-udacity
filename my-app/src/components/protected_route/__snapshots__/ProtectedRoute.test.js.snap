// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProtectedRoute Component matches snapshot when user is authenticated 1`] = `
<DocumentFragment>
  <div>
    Mock Child Content
  </div>
</DocumentFragment>
`;

exports[`ProtectedRoute Component matches snapshot when user is not authenticated 1`] = `<DocumentFragment />`;
